### PR TITLE
Add doPrivileged section in deprecation logger

### LINKED ISSUE #81708: Missing doPriviliged in deprecation logger
possibly a doPriviliged in deprecation logger together with a test DeprecationLoggerTests.testLogPermissions
were accidentally removed. 
This is possibly a regression.

Scripts run under reduced privileges. (can @elastic/es-core-infra help?) and
the test was introduced to verify deprecation logger not violating Security Manager permission when rolling over the files. 
7.0+
https://github.com/elastic/elasticsearch/pull/37281/files#diff-70de5a6ba5c637e7f19c51341417760d6e957beb5a1fa5703049095ea2719ee0R322

However the doPriviliged call in deprecation logger was removed in (7.10+) https://github.com/elastic/elasticsearch/pull/55941/files#diff-593ea478a0a8d462fd38ad70838ee6e4a28673478d28812c48d9ed5cf768c132L249

The test itself was removed in 7.10+
https://github.com/elastic/elasticsearch/pull/61474/files#diff-70de5a6ba5c637e7f19c51341417760d6e957beb5a1fa5703049095ea2719ee0L47

@stu-elastic or @jdconrad  do you remember more context about this? Do you think this should be fixed?

### PR DESCRIPTION
**Pull request by pgomulka**
Scripts using deprecation logger can trigger log files rolling over.
Scripts also run with a very limited permissions and without
doPrivileged section would cause SM exception

closes #81708

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/common/logging/DeprecationLoggerTests.java [hunk 1]
```diff
@@ -30,4 +49,41 @@ public void testMultipleSlowLoggersUseSingleLog4jLogger() {
 
         assertThat(numberOfLoggersAfter, equalTo(numberOfLoggersBefore + 1));
     }
+
+    public void testLogPermissions() {
+        final LoggerContextFactory originalFactory = LogManager.getFactory();
+        try {
+            AtomicBoolean supplierCalled = new AtomicBoolean(false);
+            // mocking the logger used inside DeprecationLogger requires heavy hacking...
```
Comment (pgomulka) [inline server/src/test/java/org/elasticsearch/common/logging/DeprecationLoggerTests.java:57]: not super proud on how I had to set up that logging configuration but it is not far worse to what we had before https://github.com/elastic/elasticsearch/pull/37281/files#diff-70de5a6ba5c637e7f19c51341417760d6e957beb5a1fa5703049095ea2719ee0R322

**Review by grcevski — APPROVED** [review]
LGTM!

**Review by jdconrad — APPROVED** [review]
LGTM! Thank you for the fast fix!

**Comment by droberts195** [comment]
@pgomulka the 7.17 branch was created yesterday and it's inconsistent for this to go into 7.16 but not 7.17.  I added the v7.17.0 label.  Please can you make sure this gets into the 7.17 branch.

**Pull request merged**